### DISCUSSION TITLE
Deploy to Web?

### DISCUSSION THREAD

**Discussion by C4ArtZ** [massCodeIO/massCode#343] [Ideas]
Are there any plans to make massCode deployable with a normal webserver?
From my understanding Electron is basicly a wrapper for a website.

I guess editing snippets with multiple users at the same time would be kinda wonky.

I love massCode so far but I would like to access my snippets from other devices without having to install the app everywhere and sync my snippets via Google Drive. I am aware that solutions like snibox exist, but I just like the feel of massCode better.

PS: I hope this is the right category 😄

**Comment by antonreshetov** [comment]
@C4ArtZ 

> From my understanding Electron is fundamentally a wrapper for a website.

That's not entirely true, Electron uses web technology, but that's only part of it. 

The core ideology of massCode is to be a free tool. Making a web version is not difficult, but you have to understand that it requires a system of accounts, data storage and so on. And this will not be a free product for the user. Let's say a typical price for the cloud version is around $5/month. 

If there are enough requests from users, then I can implement this idea.

**Reply by schwimmcoder** [reply]
> The core ideology of massCode is to be a free tool. Making a web version is not difficult, but you have to understand that it requires a system of accounts, data storage and so on. And this will not be a free product for the user. Let's say a typical price for the cloud version is around $5/month.
> 
> If there are enough requests from users, then I can implement this idea.

I would love to see massCode as an web tool for selfhosting. Would have the huge advantage to have everything synced between all devices, if I can rum massCode on my own Server.

**Reply by antonreshetov** [reply]
@schwimmcoder Running massCode on your own server is an idea that won't work without cloud storage for snippets. Cloud storage is essentially a separate premium feature that needs to be implemented and for which I will have to charge money (of course, I want to earn money 😄 ).

**Reply by schwimmcoder** [reply]
Yeah, but I assume, massCode does not need several gigabytes of cloud storage and that should be in charge of those, who would selfhost. My own server runs several docker container, so massCode would be just one more, nothing else.

**Comment by hypercube33** [comment]
I keep mine in my onedrive, but I threw a feature request in to have it support a team database and merge them or something like that. Seems to handle it already fairly well and wouldn't need a lot of code.

**Comment by craftzneko** [comment]
Would love to see this happen, I have a few docker containers running on Azure Web Apps

**Comment by snoweuph** [comment]
+1 to this.
I host most tools that I use in docker containers with SSO auth before them. I'm currently on Snippbox, but it's quite limiting and doesn't really have a good api, and thus isn't usable for my needs.

**Comment by codebude** [comment]
I also love massCode, but was missing a web viewer. Not editing, no fully-fledged replacement, but simply having the possibility to view my snippet collection from everywhere/every client (e.g. at my colleagues desks). Thus I wrote a small tool that allows to view massCode dbs in any webbrowser. Probably this may help the one or other of you. You can find it at: https://github.com/codebude/takecode

@antonreshetov I really appreciate your work. If you're not happy posting this link here, let me know or simply delete my comment.
